AUTHORING_MANIFESTS.md

```diff
@@ -77,3 +86,135 @@ It is important to test your manifest before submission to ensure it meets the r
+
+### Metadata 
+
+- DisplayName
+- DisplayVersion
+  - Displayed in the Control Panel's Add & Remove Programs page (ARP) and/or Apps and features Settings page (ANF). 
+  - Should not be used if it's always equal to PackageVersion.
+  - Must be unique per package. if multiple versions are sharing the same DisplayVersion, we may only keep the 'latest' one in the repo.
+- Package Familyname (MSIX)
+- ProductCode & Upgradecode (MSI)
+  - The ProductCode can be at any level, but UpgradeCode can only be used in AppsAndFeaturesEntries. 
+
+### 4 main use cases: 
+
+- Package does not write DisplayVersion to registry.
+  - DisplayName contains version. DisplayName is required in every manifest, to match installed packages to available versions.
+  - ProductCode contains version. ProductCode is required in every manifest, to match installed packages to available versions.
+- PackageVersion differed from DisplayVersion at any point in the manifest history. PackageVersion is required in every manifest, to prevent version range mapping errors.
+- ProductCode of the installed product differs from the ProductCode of the installer. ProductCode is required in at least one manifest to match installed packages to available packages.
+
+Note: Having incorrect values may have adverse effects, so these shouldn't be used unless there are reports of matching issues.
+
+## Sorting and sort order
+
+As long as a PackageVersion sticks to either semantic or string, it's pretty easy to sort. But switching usually causes the order to become confused. Insofar as being YAML compliant, these are all supposed to be strings, leading to a few manifests having their PackageVersion numbers nested in quotes. 
+
+### Semantic
+
+If composed solely of numbers and dots, the PackageVersion will generally be interpreted as a semantic or similar type of version. 
+
+- A "version parameter" is a set of numbers between dots in a version number - major, minor, patch, build, etc. 
+- Digits are sorted within a version parameter, with further-left parameters having magnitude over further-right parameters. 
+  - This might disrupt some temporal versioning schemae, such as `DD.MM.YYYY`, as a version from 31 Mar 2023 (`31.03.2023`) would be sorted as newer than one from 1 Feb 2024 (`01.02.2024`). 
+
+### String
+
+If any letters, spaces, or other non-numeric characters (other than dots) are added, the PackageVersion will fallback to string interpretation. 
+
+- Digits are sorted independently, and version `3.9.E` would be sorted higher than `3.10.B`. because 9 is larger than 1. This system doesn't have the concept of version parameters. 
```

Comment:
This is not how winget parses versions. Winget breaks at each `.` and compares each part individually. Where a part contains an numeral-alpha, the numeral is considered before the alpha.

For your example, the first comparison results in `3 == 3`, so then the second part is evaluated, `10 > 9`, so `3.10.B` is greater than `3.9.E`. The comparison returns and the third part is never evaluated.

I suggest reading through the commented code over at winget-cli for more detail on the nuances of how versions are broken into Parts, and Parts are broken into "integer" and "other"
https://github.com/microsoft/winget-cli/blob/8a006549c4aa0dd13cc17f6185ce30a1f4c2e71b/src/AppInstallerSharedLib/Public/AppInstallerVersions.h#L19